Modules/Loadable/Segmentations/Widgets/qMRMLSegmentsTableView.cxx

```diff
@@ -215,6 +215,14 @@ class qMRMLSegmentsTableViewPrivate: public Ui_qMRMLSegmentsTableView
 
   /// The settings key used to specify whether standard terminologies are used for name and color.
   QString TerminologySelectorSettingsKey{"Segmentations/SegmentsTableUseStandardTerminology"};
+
+  /// Counter for the number of times the user specifies custom name. Offer not using terminology if used often.
+  unsigned int NonStandardTerminologySelectedCounter{0};
```

Comment:
Sharing the counter only makes sense between segments table views that use the exact same flags and settings key. It could have been implemented by computing a hash from these variables and then storing the counters in a static QMap, but I don't think it would have improved the heuristics (it is approximate anyway) and it would have made the behavior a bit less predictable for users (they don't necessarily expect that using a segment selector in one module has an effect of how another module behaves).